main.go

```diff
@@ -36,7 +36,7 @@ import (
 var glcm = common.GetLifecycleMgr()
 
 func main() {
-	rand.Seed(time.Now().UnixNano()) // make sure our random numbers actually are random (but remember, use crypto/rand for anything where strong/reliable randomness is required
+	rand.New(rand.NewSource(time.Now().UnixNano())) // make sure our random numbers actually are random (but remember, use crypto/rand for anything where strong/reliable randomness is required
```

Comment:
This is incorrect. `rand.New` returns a new `*rand.Rand`, and does not set the seed globally. Please track down all uses of `rand` and ensure they use this `*rand.Rand`

Reply:
Actually lemme double correct this-- According to the docs, as of Go 1.20 this is no longer necessary at all. https://pkg.go.dev/math/rand#Seed

Reply:
will remove this line entirely